scipy/special/tests/test_zeta.py

```diff
@@ -47,3 +49,228 @@ def test_riemann_zeta_avoid_overflow():
     s = -260.00000000001
     desired = -5.6966307844402683127e+297  # Computed with Mpmath
     assert_allclose(sc.zeta(s), desired, atol=0, rtol=5e-14)
+
+
+@pytest.mark.parametrize(
+    "z, desired, rtol",
+    [
+        ## Test cases taken from mpmath with the script:
+
+        # import numpy as np
+        # import scipy.stats as stats
+
+        # from mpmath import mp
+
+        # # seed = np.random.SeedSequence().entropy
+        # seed = 154689806791763421822480125722191067828
+        # rng = np.random.default_rng(seed)
+        # default_rtol = 1e-13
+
+        # # A small point in each quadrant outside of the critical strip
+        # cases = []
+        # for x_sign, y_sign in [1, 1], [1, -1], [-1, 1], [-1, -1]:
+        #     x = x_sign * rng.uniform(2, 8)
+        #     y = y_sign * rng.uniform(2, 8)
+        #     z = x + y*1j
+        #     reference = complex(mp.zeta(z))
+        #     cases.append((z, reference, default_rtol))
+
+        # # Moderately large imaginary part in each quadrant outside of critical strip
+        # for x_sign, y_sign in [1, 1], [1, -1], [-1, 1], [-1, -1]:
+        #     x = x_sign * rng.uniform(2, 8)
+        #     y = y_sign * rng.uniform(50, 80)
+        #     z = x + y*1j
+        #     reference = complex(mp.zeta(z))
+        #     cases.append((z, reference, default_rtol))
+
+        # # points in critical strip
+        # x = rng.uniform(0.0, 1.0, size=5)
+        # y = np.exp(rng.uniform(0, 5, size=5))
+        # z = x + y*1j
+        # for t in z:
+        #     reference = complex(mp.zeta(t))
+        #     cases.append((complex(t), reference, default_rtol))
+        # z = x - y*1j
+        # for t in z:
+        #     reference = complex(mp.zeta(t))
+        #     cases.append((complex(t), reference, default_rtol))
+
+        # # Near small trivial zeros
+        # x = np.array([-2, -4, -6, -8])
+        # y = np.array([1e-15, -1e-15])
+        # x, y = np.meshgrid(x, y)
+        # x, y = x.ravel(), y.ravel()
+        # z = x + y*1j
+        # for t in z:
+        #     reference = complex(mp.zeta(t))
+        #     cases.append((complex(t), reference, 1e-7))
+
+        # # Some other points near real axis
+        # x = np.array([-0.5, 0, 0.2, 0.75])
+        # y = np.array([1e-15, -1e-15])
+        # x, y = np.meshgrid(x, y)
+        # x, y = x.ravel(), y.ravel()
+        # z = x + y*1j
+        # for t in z:
+        #     reference = complex(mp.zeta(t))
+        #     cases.append((complex(t), reference, 1e-7))
+
+        # # Moderately large real part
+        # x = np.array([49.33915930750887, 50.55805244181687])
+        # y = rng.uniform(20, 100, size=3)
+        # x, y = np.meshgrid(x, y)
+        # x, y = x.ravel(), y.ravel()
+        # z = x + y*1j
+        # for t in z:
+        #     reference = complex(mp.zeta(t))
+        #     cases.append((complex(t), reference, default_rtol))
+
+        # # Very large imaginary part
```

Comment:
How long do these take for you to run?

Reply:
Extremely fast

```
In [2]: %timeit run zeta_cases.py
23.1 ms ± 136 μs per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

Note sure what's going on. My mpmath version is `1.3.0` and I just use the default `dps`.